pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -203,10 +235,78 @@ func (o *MustGatherOptions) completeImages() error {
 		}
 		o.Images = append(o.Images, image)
 	}
+	if o.AllImages {
+		// find all csvs and clusteroperators with the annotation "operators.openshift.io/must-gather-image"
+		var pluginImages = map[string]struct{}{}
+
+		images, err := o.annotatedCSVs()
+		if err != nil {
+			return err
+		}
+		for k := range images {
+			pluginImages[k] = images[k]
+		}
+
+		images, err = o.annotatedCOs()
+		if err != nil {
+			return err
+		}
+		for k := range images {
+			pluginImages[k] = images[k]
+		}
+
+		for i := range pluginImages {
+			o.Images = append(o.Images, i)
+		}
+	}
 	o.log("Using must-gather plug-in image: %s", strings.Join(o.Images, ", "))
 	return nil
 }
 
+func (o *MustGatherOptions) annotatedCSVs() (map[string]struct{}, error) {
+	var annotation = "operators.openshift.io/must-gather-image"
+	var csvGVR = schema.GroupVersionResource{
+		Group:    "operators.coreos.com",
+		Version:  "v1alpha1",
+		Resource: "clusterserviceversions",
+	}
+	var pluginImages = map[string]struct{}{}
+
+	csvs, err := o.DynamicClient.Resource(csvGVR).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, item := range csvs.Items {
+		ann := item.GetAnnotations()
+		if v, ok := ann[annotation]; ok {
+			pluginImages[v] = struct{}{}
+		} else {
+			continue
+		}
+	}
+	return pluginImages, nil
+}
+
+func (o *MustGatherOptions) annotatedCOs() (map[string]struct{}, error) {
```

Comment:
I mean there are a few lines of code and it doesn't do anything fancy, we can take these lines into the caller and remove `annotatedCOs` function

Reply:
Yeah, I agree. It just feels cleaner this way. But then you are the maintainer here, so I'll go with your suggestion. :)

Reply:
Removed. :)